lib/dialects/abstract/query-generator.js

```diff
@@ -1230,14 +1230,28 @@ module.exports = (function() {
         var mainQueryOrder = [];
         var subQueryOrder = [];
 
+        var validateOrder = function(order) {
+          if (!_.contains(['ASC', 'DESC'], order.toUpperCase())) {
+            throw new Error(util.format('Order must be \'ASC\' or \'DESC\', \'%s\' given', order));
+          }
+        };
+
         if (Array.isArray(options.order)) {
           options.order.forEach(function(t) {
+            if (Array.isArray(t)) {
+              validateOrder(_.last(t));
+            }
+
             if (subQuery && (Array.isArray(t) && !(t[0] instanceof Model) && !(t[0].model instanceof Model))) {
               subQueryOrder.push(this.quote(t, model));
             }
             mainQueryOrder.push(this.quote(t, model));
           }.bind(this));
         } else {
+          if (Array.isArray(options.order)) {
```

Comment:
This condition will never be true. When the `order` is sent as a string, we'll probably need to parse it and look of the *ASC*/*DESC* keywords no?